src/widget/findonsoundcloud.cpp

```diff
@@ -0,0 +1,50 @@
+#include "findonsoundcloud.h"
+
+#include <QDesktopServices>
+#include <QMenu>
+#include <QScreen>
+#include <QUrl>
+#include <QUrlQuery>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/parented_ptr.h"
+
+namespace {
+const QString kServiceTitle = QStringLiteral("Soundcloud");
+
+const QString kSearchUrlArtist = QStringLiteral("https://soundcloud.com/search/people?");
+
+const QString kSearchUrlTitle = QStringLiteral("https://soundcloud.com/search/sounds?");
+
+const QString kSearchUrlAlbum = QStringLiteral("https://soundcloud.com/search/albums?");
+} // namespace
+
+void FindOnSoundcloud::addSubmenusForServices(QMenu* pFindOnMenu, const Track& track) {
+    const auto artist = track.getArtist();
+    const auto trackTitle = track.getTitle();
+    const auto album = track.getAlbum();
+    auto m_pServiceMenu = make_parented<QMenu>(this);
+    m_pServiceMenu->setTitle(kServiceTitle);
+    pFindOnMenu->addMenu(m_pServiceMenu);
+    addSeparator();
+    if (!artist.isEmpty()) {
+        m_pServiceMenu->addAction(
```

Comment:
I want to addActions, with the signal triggerbrowser. But In order to do that I think I need an object to call from. I actually created an action with the commented line, triggerBrowser was a signal from FindOnWeb class. But then I couldn't connect it from WTrackMenu. 

I have searched about it such as factory pattern with QT signal, or signal from sub classes etc. Couldn't find a proper solution. If you could give me an documents, example or any other resource I would like to search and do.

I wouldn't want to commit before everything is done, but then I realised I lost a lot of time with this, I will work on the other PR meanwhile.